scripts/generate_doc_stubs.py

```diff
@@ -0,0 +1,590 @@
+"""
+Generate API documentation for AWS SDK Python clients.
+
+This script generates MkDocs Material documentation that matches the reference
+documentation format. It analyzes Python client packages using griffe
+and creates structured documentation with proper formatting.
+"""
+
+import logging
+import sys
+from dataclasses import dataclass, field
+from enum import Enum
+from pathlib import Path
+
+import griffe
+import mkdocs_gen_files
+from griffe import Alias, Class, Function, Module, Object, ExprBinOp, ExprSubscript, TypeAlias
+
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="[%(asctime)s - %(name)s - %(levelname)s] %(message)s",
+    datefmt="%Y-%m-%d %H:%M:%S",
+)
+logger = logging.getLogger("generate_doc_stubs")
+
+EVENT_STREAM_TYPES = ["InputEventStream", "OutputEventStream", "DuplexEventStream"]
+
+class StreamType(Enum):
+    """Type of event stream for operations."""
+
+    NONE = "none"
+    INPUT = "InputEventStream"  # Client-to-server streaming
+    OUTPUT = "OutputEventStream"  # Server-to-client streaming
+    DUPLEX = "DuplexEventStream"  # Bidirectional streaming
+
+ 
+@dataclass(frozen=True)
+class StructureInfo:
+    """Information about a structure (dataclass)."""
+
+    name: str
+    module_path: str  # e.g., "aws_sdk_bedrock_runtime.models"
+
+
+@dataclass
+class UnionInfo:
+    """Information about a union type."""
+
+    name: str
+    module_path: str
+    members: list[StructureInfo]
+
+
+@dataclass
+class EnumInfo:
+    """Information about an enum."""
+
+    name: str
+    module_path: str
+
+
+@dataclass
+class ErrorInfo:
+    """Information about an error/exception."""
+
+    name: str
+    module_path: str
+
+
+@dataclass
+class ConfigInfo:
+    """Information about a configuration class."""
+
+    name: str
+    module_path: str
+
+
+@dataclass
+class PluginInfo:
+    """Information about a plugin type alias."""
+
+    name: str
+    module_path: str
+
+
+@dataclass
+class OperationInfo:
+    """Information about a client operation."""
+
+    name: str  # e.g., "converse"
+    module_path: str # e.g., "aws_sdk_bedrock_runtime.client.BedrockRuntimeClient.converse"
+    input: StructureInfo # e.g., "ConverseInput"
+    output: StructureInfo # e.g., "ConverseOperationOutput"
+    stream_type: StreamType.NONE
```

Comment:
This either needs to be just `StreamType` or `StreamType = StreamType.NONE` if you meant to set a default